generic/tests/boot.py

```diff
@@ -18,15 +19,56 @@ def run(test, params, env):
     :param env: Dictionary with test environment.
     """
 
+    def get_valid_nic(vm):
+        """
+        return first nic which has an IP address assigned system
+        :param vm: vm object
+        :return: nic_index which is the int type(e.g. 0, 1, 2.) or None
+        """
+        nic_number_list = params.objects("nics")
+        for idx in range(len(nic_number_list)):
+            if vm.wait_for_get_address(idx):
+                return idx
+        return None
```

Comment:
If no valid NIC, I think the case should fail/cancel. Because `vm.wait_for_login(nic_index=None, timeout=timeout)` will directly trigger an exception.

Reply:
> If no valid NIC, I think the case should fail/cancel. Because `vm.wait_for_login(nic_index=None, timeout=timeout)` will directly trigger an exception.

fail statement will be updated at here.